From ad5347242c336eb63465d00c763f2abebd0b6c65 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Tue, 31 May 2016 08:05:53 +0000
Subject: [PATCH] wr-policy: fix avc denials for tcsd

* allow tcsd_t to read files at all levels

Fix avc denials:

  avc: denied { read } for pid=7943 comm="tcsd" \
  name="ld.so.cache" dev="dm-1" ino=31285 \
  scontext=system_u:system_r:tcsd_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:etc_t:s15:c0.c1023 \
  tclass=file permissive=0

  avc: denied { search } for pid=7943 comm="tcsd" \
  name="journal" dev="tmpfs" ino=67 \
  scontext=system_u:system_r:tcsd_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:syslogd_var_run_t:s15:c0.c1023 \
  tclass=dir permissive=0

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/contrib/tcsd.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/contrib/tcsd.te b/policy/modules/contrib/tcsd.te
index ca98bf8..30033ca 100644
--- a/policy/modules/contrib/tcsd.te
+++ b/policy/modules/contrib/tcsd.te
@@ -53,3 +53,5 @@ init_read_utmp(tcsd_t)
 logging_send_syslog_msg(tcsd_t)
 
 miscfiles_read_localization(tcsd_t)
+
+mls_file_read_all_levels(tcsd_t)
-- 
2.8.1

